epsagon/modules/tornado.py

```diff
@@ -229,6 +235,58 @@ def thread_pool_execution(cls, unique_id, fn, args, kwargs):
         return res
 
 
+def _prepare_http_request(request, raise_error=True, **kwargs):
+    """
+    Copying parameters from original `AsyncHTTPClient.fetch` function
+    :return: HTTPRequest, raise_error bool
+    """
+    # request can be a URL string or a HTTPRequest object
+    if not isinstance(request, HTTPRequest):
+        request = HTTPRequest(request, **kwargs)
+
+    return request, raise_error
+
+
+def _wrapper(wrapped, instance, args, kwargs):
+    """
+    General wrapper for AsyncHTTPClient instrumentation.
+    :param wrapped: wrapt's wrapped
+    :param instance: wrapt's instance
+    :param args: wrapt's args
+    :param kwargs: wrapt's kwargs
+    :return: None
+    """
+    try:
+        request, raise_error = _prepare_http_request(*args, **kwargs)
+    except Exception:  # pylint: disable=W0703
+        return wrapped(*args, **kwargs)
+
+    # Inject header to support tracing over HTTP requests
+    trace_id = uuid.uuid4().hex
```

Comment:
I'd maybe export this header creation logic somewhere since we do it everywhere now

Reply:
Done